selfservice/hook/web_hook.go

```diff
@@ -502,15 +502,16 @@ func parseResponse(l *logrusx.Logger, resp *http.Response) (err error) {
 	}
 
 	body, err := io.ReadAll(resp.Body)
-	if err != nil {
-		return errors.Wrap(err, "could not read response body")
-	}
 	defer func(Body io.ReadCloser) {
 		if closeErr := Body.Close(); closeErr != nil {
 			err = closeErr
 		}
 	}(resp.Body)
```

Comment:
Is Body pointer for sure? If not we are making the Copy of it and therefore not closing correct instance of io.ReaderCloser and this can cause memory leak

Reply:
`io.ReadCloser` is an interface and `Response.Body` is also an interface. We are passing pointers here.